src/docs/arch-checkpoint.dox

```diff
@@ -19,46 +21,71 @@ WT_CONNECTION::rollback_to_stable.
 
 # The checkpoint algorithm #
 
-A checkpoint can be broken up into 5 main stages:
+The checkpoint algorithm can be considered as a coordinator of multiple aspects of the system. It
+relies on the @ref arch-btree "btree subsystem" for traversing in memory pages, the @ref arch-block "block manager subsystem" for
+handling on-disk blocks and the @ref arch-metadata "metadata subsystem" for constructing and interpreting the metadata
+information.
 
-_The prepare stage:_
+A checkpoint can be broken up into the following stages:
+
+## Locking ##
+
+The first stage of checkpoint is to gather the necessary checkpoint locks. The most important being
+the \c checkpoint_lock. This lock ensures only one checkpoint takes place at time. More granular locks
+such as, dhandle, schema and flush locks are taken throughout the checkpoint process.
+
+## Eviction ##
+
+Before the checkpoint transaction begins (pinning content), the @ref arch-eviction "eviction server" is leveraged to
+reduce the dirty content in the cache. This has the benefit of utilizing the multi-threaded
```

Comment:
"This has..." the meaning of "this" is ambiguous. Maybe "Using eviction has the benefit of taking advantage of the multi-threaded..."

Reply:
Good point!

Reply:
https://github.com/wiredtiger/wiredtiger/pull/11576/commits/206ef8b5c0edfa3cdf753e26542f27d46eaf1f14